engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateByWindowTime.java

```diff
@@ -338,9 +340,9 @@ public void processRows(UpdateByWindowBucketContext context, boolean initialStep
                 final WritableIntChunk<? extends Values> popChunk =
                         WritableIntChunk.makeWritableChunk(ctx.workingChunkSize)) {
 
-            long currentHeadTs = nextLongOrMax(influencerTsHeadIt);
             long currentTailTs = nextLongOrMax(influencerTsTailIt);
 
+
```

Comment:
We talked about re-writing this loop with:
```
        try (final RowSequence.Iterator affectedRowsIt = ctx.affectedRows.getRowSequenceIterator();
                final LongColumnIterator affectedTsIt =
                        new LongColumnIterator(context.timestampColumnSource, ctx.affectedRows);
                final LongColumnIterator influencerTsTailIt =
                        new LongColumnIterator(context.timestampColumnSource, context.influencerRows);
                final RowSequence.Iterator influencerRowsIt = ctx.influencerRows.getRowSequenceIterator();
                final WritableIntChunk<? extends Values> pushChunk =
                        WritableIntChunk.makeWritableChunk(ctx.workingChunkSize);
                final WritableIntChunk<? extends Values> popChunk =
                        WritableIntChunk.makeWritableChunk(ctx.workingChunkSize)) {

            long currentTailTs = nextLongOrMax(influencerTsTailIt);
            long totalPushCount = 0;

            while (affectedRowsIt.hasMore()) {
                // NOTE: we did not put null values into our SSA and our influencer rowset is built using the
                // SSA. there should be no null timestamps considered in the rolling windows

                int affectedRowIndex;
                for (affectedRowIndex = 0;
                     affectedRowIndex < ctx.workingChunkSize && affectedTsIt.hasNext();
                     affectedRowIndex++) {
                    // read the current timestamp
                    final long currentTimestamp = affectedTsIt.nextLong();
                    if (currentTimestamp == NULL_LONG) {
                        // this signifies that does not belong to a time window
                        popChunk.set(affectedRowIndex, NULL_INT);
                        pushChunk.set(affectedRowIndex, NULL_INT);
                        continue;
                    }

                    // compute the head and tail timestamps (inclusive)
                    final long head = currentTimestamp - prevUnits;
                    final long tail = currentTimestamp + fwdUnits;

                    // pop out all values from the current window that are not in the new window
                    long popCount = 0;
                    while (!ctx.timestampWindowBuffer.isEmpty() && ctx.timestampWindowBuffer.front() < head) {
                        ctx.timestampWindowBuffer.remove();
                        popCount++;
                    }

                    // advance the keyIt and timestamp iterators until we are within the window. This only happens
                    // when initialStep == true because we have not created the minimum set of rows but include all
                    // non-null timestamp rows in our influencer values
                    long skipCount = 0;
                    while (currentTailTs < head) {
                        Assert.eqTrue(initialStep, "initialStep when skipping rows");
                        currentTailTs = nextLongOrMax(influencerTsTailIt);
                        skipCount++;
                    }
                    influencerRowsIt.getNextRowSequenceWithLength(skipCount);

                    // push in all values that are in the new window (inclusive of tail)
                    long pushCount = 0;
                    while (currentTailTs <= tail) {
                        // add this value to the buffer before advancing
                        ctx.timestampWindowBuffer.add(currentTailTs);
                        currentTailTs = nextLongOrMax(influencerTsTailIt);
                        pushCount++;
                    }

                    // write the push and pop counts to the chunks
                    popChunk.set(affectedRowIndex, Math.toIntExact(popCount));
                    pushChunk.set(affectedRowIndex, Math.toIntExact(pushCount));
                    totalPushCount += pushCount;
                }
                final RowSequence chunkAffectedRows = affectedRowsIt.getNextRowSequenceWithLength(affectedRowIndex);
                final RowSequence chunkInfluencerRows = influencerRowsIt.getNextRowSequenceWithLength(totalPushCount);
                // execute the operators
                ensureGetContextSize(ctx, chunkInfluencerRows.size());

                Arrays.fill(ctx.inputSourceChunks, null);
                for (int opIdx : context.dirtyOperatorIndices) {
                    UpdateByWindowedOperator.Context opCtx =
                            (UpdateByWindowedOperator.Context) context.opContext[opIdx];
                    // prep the chunk array needed by the accumulate call
                    final int[] srcIndices = operatorInputSourceSlots[opIdx];
                    for (int ii = 0; ii < srcIndices.length; ii++) {
                        int srcIdx = srcIndices[ii];
                        // chunk prep
                        prepareValuesChunkForSource(ctx, srcIdx, chunkInfluencerRows);
                        opCtx.chunkArr[ii] = ctx.inputSourceChunks[srcIdx];
                    }

                    // make the specialized call for windowed operators
                    ((UpdateByWindowedOperator.Context) ctx.opContext[opIdx]).accumulate(
                            chunkAffectedRows,
                            opCtx.chunkArr,
                            pushChunk,
                            popChunk,
                            chunkAffectedRows.intSize());
                }
            }
        }
```